htmd/parameterization/dihedral.py

```diff
@@ -242,39 +253,72 @@ def _vectorToParams(self, vector):
 
         return parameters
 
-    def _optimizeWithRandomSearch(self, vector):
+    def _optimizeWithIterativeScheme(self, vector, target_energies):
+        logger.info('Using iterative fitting scheme with {} iterations'.format(self.num_iterations))
+        best_loss = None
+        best_vector = None
+        for i in range(self.num_iterations):
+            logger.info('Dihedral fitting iteration {}'.format(i+1))
+
+            # Evaluate constant terms for single dihedral fitting
+            current_params = self._vectorToParams(vector)
+            const_energies_dihedral = self._evaluateConstTermsPerDihedral(current_params)
+            target_energies_dihedral = self._calculateTargetEnergies(const_energies_dihedral)
+
+            # Fit the individual dihedrals
+            dihedral_vectors = []
+            for idihed in range(self.numDihedrals):
+                logger.info('Optimizing dihedral {}/{}'.format(idihed+1, self.numDihedrals))
+                subvector = self._paramsToVector(current_params, idihed)
+                subvector = self._optimizeWithRandomSearch(subvector, target_energies_dihedral, num_searches=20, idihed=idihed, _logger=False)
+                dihedral_vectors.append(subvector)
+
+            # Pack the individually fitted parameters into a vector
+            vector = DihedralFitting._subvectorsToVector(dihedral_vectors, vector)
+
+            # Perform global optimization of all dihedrals
+            logger.info('Globally optimizing all dihedrals')
+            vector = self._optimizeWithRandomSearch(vector, target_energies, best_vector=best_vector, best_loss=best_loss, num_searches=1)
+            best_vector = vector.copy()
+            best_loss = self._objective(vector, None, self._angle_values, target_energies)
+
+        return vector
+
+    def _optimizeWithRandomSearch(self, vector, target_energies, idihed=None, convergence_rmsd=1e-3,
+                                  num_searches=None, best_vector=None, best_loss=None, _logger=True):
         """
         Naive random search
         """
+        numDihedrals = self.numDihedrals if idihed is None else 1
 
         # Create a local optimizer
         opt = nlopt.opt(nlopt.LD_LBFGS, vector.size)
         opt.set_vector_storage(opt.get_dimension())
-        logger.info('Local optimizer: {} with {} vector storage'.format(opt.get_algorithm_name(),
```

Comment:
It's information that tells nothing to the user. He cannot change it. Also it gets printed at every individual dihedral optimization which makes for a bit of spam.

Reply:
I added back in information on the number of random searches as it might be useful to know how long it will take. The optimizer information I still don't see as of any use as I said before. We are not planning on making the NRS default anyway so it doesn't matter too much.